### DISCUSSION TITLE
Asking help on some variables in interface type 2 codes

### DISCUSSION THREAD

**Discussion by halfwitbm** [OpenRadioss/OpenRadioss#2011] [Q&A] [answered]
Hi All,

When I read interface type 2 related code, I find some variables are hard to find comment or help information in OpenRadioss source code, but they make me unable to understand code. Can somebody give me some help information on these variables ( What's these variable's name stand for). Thanks in advance. 
Variables list: NSV, NSVG, IRTL, CRST, TT, IGTYP, FSKYI2, INDXP, IRODDL.
These variables are i2for28p_cin, i2rupt.F and ruptint2.F

Regards,

Bin

**Comment by bertrand-altair** [comment, marked as answer]
Hi Bin,

Here are the definitions for the arrays you have mentioned:

NSV:  internal id of secondary node
NSVG: used only with AMS (advanced Mass Scaling) - array used to store the id of secondary node in case of penalty tied contact for computation of added mass.
IRTL: id of the main surface in IRECTL where corresponding nodes are stored
CRST:  coordinates of the projection of the secondary node in the parametric space of the main surface
TT: current time
IGTYP:  property type - no longer used - always equal to -1
FSKYI2: Array used for PARTIH/ON assembly in SPMD (used to ensure that forces on nodes of the main surface are always summed in the same order).
INDXP:  INDXC and INDXP are index arrays for kinematic (INDXC) and penalty (INDXP) tied contacts (defined in i2for28.F)
IRODDL: flag for rotational degree of freedom computation - if equal to 0 then rotational dof are not computed (typically if the model contains only solid elements)

Regards,

Bertrand,

**Reply by halfwitbm** [reply]
Hi Bertrand,

Thank you very much for the detailed explanation. 🥇 

Regards,

Bin
